olp-cpp-sdk-core/src/network2/curl/NetworkProtocolCurl.cpp

```diff
@@ -0,0 +1,1297 @@
+      curl_easy_setopt(handle->handle, CURLOPT_PROXYPASSWORD,
+                       proxy.GetPassword().c_str());
+    }
+  }
+
+  if (handle->chunk) {
+    curl_easy_setopt(handle->handle, CURLOPT_HTTPHEADER, handle->chunk);
+  }
+
+#ifdef NETWORK_HAS_OPENSSL
+  std::string curlCaBundle = ""; // config->GetCaCert();
+  if (curlCaBundle.empty()) {
+    curlCaBundle = ca_bundle_path();
+  }
+  if (!curlCaBundle.empty()) {
+    CURLcode error =
+        curl_easy_setopt(handle->handle, CURLOPT_CAINFO, curlCaBundle.c_str());
+    if (CURLE_OK != error) {
+      return ErrorCode::UNKNOWN_ERROR;//NetworkProtocol::ErrorFailedSetCaCert;
+    }
+    EDGE_SDK_LOG_TRACE(LOGTAG, "curl bundle path: " << curlCaBundle);
+  }
+#endif
+
+  if (sysDontVerifyCertificate) {
+    curl_easy_setopt(handle->handle, CURLOPT_SSL_VERIFYPEER, 0L);
+    curl_easy_setopt(handle->handle, CURLOPT_SSL_VERIFYHOST, 0L);
+  } else {
+    curl_easy_setopt(handle->handle, CURLOPT_SSL_VERIFYPEER, 1L);
+    curl_easy_setopt(handle->handle, CURLOPT_SSL_VERIFYHOST, 2L);
+#ifdef NETWORK_USE_TIMEPROVIDER
+    curl_easy_setopt(handle->handle, CURLOPT_SSL_CTX_FUNCTION, sslctx_function);
+#endif
+  }
+
+  curl_easy_setopt(handle->handle, CURLOPT_FOLLOWLOCATION, 1L);
+  curl_easy_setopt(handle->handle, CURLOPT_CONNECTTIMEOUT,
+                   config.GetConnectionTimeout());
+  curl_easy_setopt(handle->handle, CURLOPT_TIMEOUT,
+                   config.GetConnectionTimeout() + config.GetTransferTimeout());
+  curl_easy_setopt(handle->handle, CURLOPT_WRITEFUNCTION,
+                   &NetworkProtocolCurl::RxFunction);
+  curl_easy_setopt(handle->handle, CURLOPT_WRITEDATA, handle);
+  curl_easy_setopt(handle->handle, CURLOPT_HEADERFUNCTION,
+                   &NetworkProtocolCurl::HeaderFunction);
+  curl_easy_setopt(handle->handle, CURLOPT_HEADERDATA, handle);
+  curl_easy_setopt(handle->handle, CURLOPT_FAILONERROR, 0);
+  if (stderr_ == nullptr) {
+    curl_easy_setopt(handle->handle, CURLOPT_STDERR, 0);
+  }
+  curl_easy_setopt(handle->handle, CURLOPT_ERRORBUFFER, handle->error_text);
+
+#if (LIBCURL_VERSION_MAJOR >= 7) && (LIBCURL_VERSION_MINOR >= 21)
+  // if (config && config->IsAutoDecompressionEnabled()) {
+    curl_easy_setopt(handle->handle, CURLOPT_ACCEPT_ENCODING, "");
+    curl_easy_setopt(handle->handle, CURLOPT_TRANSFER_ENCODING, 1L);
+  // }
+#endif
+
+#if (LIBCURL_VERSION_MAJOR >= 7) && (LIBCURL_VERSION_MINOR >= 25)
+  // Enable keep-alive (since Curl 7.25.0)
+  curl_easy_setopt(handle->handle, CURLOPT_TCP_KEEPALIVE, 1L);
+  curl_easy_setopt(handle->handle, CURLOPT_TCP_KEEPIDLE, 120L);
+  curl_easy_setopt(handle->handle, CURLOPT_TCP_KEEPINTVL, 60L);
+#endif
+
+  if (!IsStarted()) {
+    return ErrorCode::OFFLINE_ERROR; //NetworkProtocol::ErrorNotReady;
+  }
+  {
+    std::lock_guard<std::mutex> lock(mutex_);
+    AddEvent(EventInfo::Type::SendEvent, handle);
+  }
+  return ErrorCode::NO_ERROR; //NetworkProtocol::ErrorNone;
+}  // namespace network
```

Comment:
```suggestion
}
```
It just a closing bracket for Send2 method